### PR TITLE
rustjail: get home dir using nix crate

### PR DESCRIPTION
**Pull request by justxuewei**
Get user's home dir using `nix::unistd` crate instead of `utils` crate,
and remove useless code from agent.

BTW, this is my first PR for kata containers community, please let me know if there are any problems with this PR, thanks.

Fixes: https://github.com/kata-containers/kata-containers/issues/4209

Signed-off-by: Xuewei Niu <justxuewei@apache.org>

### CODE REVIEW

**Comment by liubin** [comment]
@justxuewei Please update your commit message to match our [patch format](http://liubin.org/kata-dev-book/src/your-first-pr.html#%E6%8F%90%E4%BA%A4%E4%BB%A3%E7%A0%81)

**Comment by justxuewei** [comment]
> @justxuewei Please update your commit message to match our [patch format](http://liubin.org/kata-dev-book/src/your-first-pr.html#%E6%8F%90%E4%BA%A4%E4%BB%A3%E7%A0%81)

Thanks for your reply, I will update commit message later.

**Comment by justxuewei** [comment]
> > @justxuewei Please update your commit message to match our [patch format](http://liubin.org/kata-dev-book/src/your-first-pr.html#%E6%8F%90%E4%BA%A4%E4%BB%A3%E7%A0%81)
> 
> Thanks for your reply, I will update commit message later.

Hi, Bin,

The commit message was updated, please take a look, thanks.

**Comment by liubin** [comment]
@lifupan PTAL

File: src/agent/rustjail/src/container.rs [hunk 1]
```diff
@@ -669,7 +667,12 @@ fn do_init_child(cwfd: RawFd) -> Result<()> {
     // there's no uid entry in /etc/passwd, set "/" as the
     // home env.
     if env::var_os(HOME_ENV_KEY).is_none() {
-        let home_dir = utils::home_dir(guser.uid).unwrap_or_else(|_| String::from("/"));
+        let mut home_dir = String::from("/");
+        if let Ok(Some(user)) = User::from_uid(Uid::from_raw(guser.uid)) {
```
Comment (jodh-intel) [inline src/agent/rustjail/src/container.rs:671]: Nit: I think you could rearrange this slightly to remove the need for `mut`.

**Review by jodh-intel — APPROVED** [review]
Hi @justxuewei - lgtm!

File: src/agent/rustjail/src/container.rs (hunk 1 above)
Comment (justxuewei) [reply src/agent/rustjail/src/container.rs]: Hello @jodh-intel,

Thanks for your review, I was removed mut variable from code, PTAL :)

**Review by jodh-intel — APPROVED** [review]
Thanks @justxuewei.

lgtm

**Review by liubin — APPROVED** [review]

**Pull request merged**